roles/rsyslog/tasks/main.yaml

```diff
@@ -1,44 +1,25 @@
----
 - name: Install/Update required packages
   package:
-    name: "{{ lookup('flattened', rsyslog_base_packages, rsyslog_tls_packages if rsyslog_pki else [], rsyslog_packages) }}"
+    name: "{{ (__rsyslog_base_packages) + (__rsyslog_tls_packages if rsyslog_pki else []) + (rsyslog_additional_packages) }}"
     state: latest
```

Comment:
see the note above regarding `latest`

Reply:
I'm wondering when the system has the older rsyslog package is installed and there is a newer rsyslog in the repository, what happens if the state is present vs. latest.  Can we make sure we could install the latest package?

Reply:
It hasn't been a problem for other system roles apparently.

Reply:
Ok, it's guaranteed to be upgraded with `present`.  Reading this doc, I was not sure it was. :)
https://docs.ansible.com/ansible/latest/modules/package_module.html#parameter-state

I'm going to revert "latest" changes...

Reply:
> what happens if the state is present vs. latest

If the rsyslog package does not exist, then both `state: present` and `state: latest` will install the latest package available.

If the rsyslog package does exist, then `state: present` is a no-op, and `state: latest` will do an upgrade to the latest package available, if there is a newer one available.

Reply:
Also - if we do not hardcode `state: latest` in the role, then we need some other mechanism like https://github.com/geerlingguy/ansible-role-apache/blob/77f39119bc12fd839d16a55a19354d5c159041fa/defaults/main.yml#L58 - also note that if we do this, we should be consistent among *all* of the system roles - that is, every system role will need to have a `ROLE_packages_state: present` in the `defaults/main.yml` - we'll also need to add this to the README.md, the template, and the best practices guide.

Reply:
> > what happens if the state is present vs. latest
> If the rsyslog package does exist, then `state: present` is a no-op, and `state: latest` will do an upgrade to the latest package available, if there is a newer one available.

Thanks, @richm.  Hmmm, then I'd like to keep `latest` for rsyslog package install/upgrade.  The logic is also used for updating security related packages.  So, I think we need to make sure it updates to the latest.

Reply:
Ok, keep it for now, but we should set a common standard for all the roles, because I don't see why logging should be different.

Reply:
Thanks, @pcahyna.  Just out of curiosity, there's no requirement for other roles to _upgrade_ existing packages?

Reply:
I am not aware of such a requirement. As you can see above, the default for many popular roles on Galaxy is to not update.